src/main/java/frc/robot/subsystems/swerve/SwerveConfig.java

```diff
@@ -0,0 +1,135 @@
+
+
+    /* Module Gear Ratios */
+    public static final double driveGearRatio = chosenModule.driveGearRatio;
+    public static final double angleGearRatio = chosenModule.angleGearRatio;
+
+    // encoder setup
+    // meters per rotation
+    public static final double driveRevToMeters =  wheelCircumference / driveGearRatio;
+    public static final double driveRpmToMetersPerSecond = driveRevToMeters / 60 ;
+    // the number of degrees that a single rotation of the turn motor turns the wheel.
+    public static final double DegreesPerTurnRotation = 360/angleGearRatio;
+
+    
+    /* Motor Inverts */
+    public static final boolean angleMotorInvert = chosenModule.angleMotorInvert;
+    public static final boolean driveMotorInvert = chosenModule.driveMotorInvert;
+
+    /* Angle Encoder Invert */
+    public static final boolean canCoderInvert = chosenModule.canCoderInvert;
+
+    /* Swerve Current Limiting */
+    public static final int angleContinuousCurrentLimit = 40;
+    public static final int anglePeakCurrentLimit = 40; // Not in use
+    public static final double anglePeakCurrentDuration = 0.1; // Not in use
+    public static final boolean angleEnableCurrentLimit = true; // Not in use
+
+    public static final int driveContinuousCurrentLimit = 40; 
+    public static final int drivePeakCurrentLimit = 60; // Not in use
+    public static final double drivePeakCurrentDuration = 0.1; // Not in use
+    public static final boolean driveEnableCurrentLimit = true; // Not in use
+
+    /* These values are used by the drive falcon to ramp in open loop and closed loop driving.
+     * We found a small open loop ramp (0.25) helps with tread wear, tipping, etc */
+    public static final double openLoopRamp = 0.25;
+    public static final double closedLoopRamp = 0.0;
+
+    /* Angle Motor PID Values */
+    public static final double angleKP = 0.2; // 6.0, 0.1
+    public static final double angleKI = 0.0;
+    public static final double angleKD = 0.15;
+
+    // public static final double angleKP = 0.0; // 6.0, 0.1
+    // public static final double angleKI = 0.0;
+    // public static final double angleKD = 0.0;
+
+    public static final double angleKF = 0;
+
+    /* Drive Motor PID Values */
+    public static final double driveKP = 0.035; 
+    public static final double driveKI = 0.0;
+    public static final double driveKD = 0.0;
+    public static final double driveKF = 0.05; // try 0.1 or 0.13 later on 
+
+
+    /* Drive Motor Characterization Values 
+     * Divide SYSID values by 12 to convert from volts to percent output for CTRE */
+    public static final double driveKS = (0.1); 
+    public static final double driveKV = (0.1);
+    public static final double driveKA = (0.05);
+
+    /* Swerve Profiling Values */
+    public static final double driveBaseRadius = Math.hypot(trackWidth / 2.0, trackWidth / 2.0);
+    /** Meters per Second */
+    public static final double maxSpeed = Units.feetToMeters(17.1); // 4.0
```

Comment:
Is this a desired speed or a calculated speed?  Its use in SwerveMod.setSpeed suggests it is a calculated value (how fast the robot actually moves when the motor is set to its max limits)

Reply:
Yes, it is the calculated max speed - it's the same value from last year, however.